crates/stages/src/stages/sender_recovery.rs

```diff
@@ -188,14 +186,11 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
 }
 
 fn recover_sender(
-    entry: Result<(RawKey<TxNumber>, RawValue<TransactionSignedNoHash>), DatabaseError>,
+    (tx_id, tx): (TxNumber, RawValue<TransactionSignedNoHash>),
     rlp_buf: &mut Vec<u8>,
 ) -> Result<(u64, Address), Box<SenderRecoveryStageError>> {
-    let (tx_id, transaction) =
-        entry.map_err(|e| Box::new(SenderRecoveryStageError::StageError(e.into())))?;
-    let tx_id = tx_id.key().expect("key to be formated");
+    let tx = tx.value().expect("value to be formated");
```

Comment:
for example here, could it be possible that one tx value is not formatted but other tx values are? in that case, it's better not to panic

Reply:
alright, in case of a bug, would be nice to get more context, same advice you gave me last week

Reply:
btw, here, and in tracing, it's nice if the variable names match the log message keys exactly. makes it easier to debug. so, 
```rust
let my_var = 1;
debug!(target: "..",
    my_var=my_var,
    ""
);
```